Doc/c-api/init.rst

```diff
@@ -2339,6 +2345,23 @@ code triggered by the finalizer blocks and calls :c:func:`PyEval_SaveThread`.
 
    .. versionadded:: 3.13
 
+.. c:macro:: Py_BEGIN_CRITICAL_SECTION_MUTEX(m)
+
+   Locks the mutex *m* and begins a critical section.
+
+   In the free-threaded build, this macro expands to::
+
+     {
+          PyCriticalSection _py_cs;
+          PyCriticalSection_BeginMutex(&_py_cs, m)
+
+   Note that unlike :c:macro:`Py_BEGIN_CRITICAL_SECTION`, there is no cast for
+   the second argument - it must be a :c:type:`PyMutex`.
+
+   On the default build, this macro expands to ``{``.
+
+   .. versionadded:: 3.15
```

Comment:
Per @vstinner, I think we use `versionadded:: next` here but I'm not entirely sure why.

```suggestion
   .. versionadded:: next
```